Sources/SwiftSyntaxMacroExpansion/MacroExpansion.swift

```diff
@@ -295,6 +302,39 @@ public func expandAttachedMacroWithoutCollapsing<Context: MacroExpansionContext>
         return "extension \(typeName) : \(protocolName) \(whereClause) {}"
       }
 
+    case (let attachedMacro as ExtensionMacro.Type, .extension):
+      guard let declGroup = declarationNode.asProtocol(DeclGroupSyntax.self) else {
+        // Compiler error: type mismatch.
+        throw MacroExpansionError.declarationNotDeclGroup
+      }
+
+      guard let extendedType = extendedType else {
+        throw MacroExpansionError.noExtendedTypeSyntax
+      }
+
+      // Local function to expand a extension macro once we've opened up
+      // the existential.
+      func expandExtensionMacro(
+        _ node: some DeclGroupSyntax
+      ) throws -> [ExtensionDeclSyntax] {
+        return try attachedMacro.expansion(
+          of: attributeNode,
+          attachedTo: node,
+          providingExtensionsOf: extendedType,
+          in: context
+        )
+      }
+
+      let extensions = try _openExistential(
```

Comment:
I suspect we don't need to do `_openExistential` tricks nowadays, but it's fine this way